src/proto/media/file/mod.rs

```diff
@@ -115,14 +112,20 @@ impl<'a> File<'a> {
     /// * `uefi::Status::DEVICE_ERROR`       The device reported an error, the file was deleted,
     ///                                      or the end of the file was reached before the `read()`.
     /// * `uefi::Status::VOLUME_CORRUPTED`   The filesystem structures are corrupted
-    /// * `uefi::Status::BUFFER_TOO_SMALL`   The buffer is too small to hold a directory entry.
-    pub fn read(&mut self, buffer: &mut [u8]) -> result::Result<usize, (Status, usize)> {
+    /// * `uefi::Status::BUFFER_TOO_SMALL`   The buffer is too small to hold a directory entry,
+    ///                                      and the required buffer size is provided as output.
+    pub fn read(&mut self, buffer: &mut [u8]) -> Result<usize, Option<usize>> {
```

Comment:
In case you are wondering why read's API specification is so different from that of write, that is a sad result of UEFI conflating together file readout and directory entry readout.